pandas_gbq/load.py

```diff
@@ -52,45 +54,110 @@ def split_dataframe(dataframe, chunksize=None):
         yield remaining_rows, chunk
 
 
-def load_chunks(
-    client,
-    dataframe,
-    destination_table_ref,
-    chunksize=None,
-    schema=None,
-    location=None,
+def load_parquet(
+    client: bigquery.Client,
+    dataframe: pandas.DataFrame,
+    destination_table_ref: bigquery.TableReference,
+    location: Optional[str],
+    schema: Optional[Dict[str, Any]],
+):
+    job_config = bigquery.LoadJobConfig()
+    job_config.write_disposition = "WRITE_APPEND"
+    job_config.source_format = "PARQUET"
+
+    if schema is not None:
+        schema = pandas_gbq.schema.remove_policy_tags(schema)
+        job_config.schema = pandas_gbq.schema.to_google_cloud_bigquery(schema)
+
+    client.load_table_from_dataframe(
+        dataframe, destination_table_ref, job_config=job_config, location=location,
+    ).result()
+
+
+def load_csv(
+    client: bigquery.Client,
+    dataframe: pandas.DataFrame,
+    destination_table_ref: bigquery.TableReference,
+    location: Optional[str],
+    chunksize: Optional[int],
+    schema: Optional[Dict[str, Any]],
 ):
     job_config = bigquery.LoadJobConfig()
     job_config.write_disposition = "WRITE_APPEND"
     job_config.source_format = "CSV"
     job_config.allow_quoted_newlines = True
 
-    # Explicit schema? Use that!
     if schema is not None:
         schema = pandas_gbq.schema.remove_policy_tags(schema)
         job_config.schema = pandas_gbq.schema.to_google_cloud_bigquery(schema)
-    # If not, let BigQuery determine schema unless we are encoding the CSV files ourselves.
-    elif not FEATURES.bigquery_has_from_dataframe_with_csv:
+
+    chunks = split_dataframe(dataframe, chunksize=chunksize)
+    for remaining_rows, chunk in chunks:
+        yield remaining_rows
+
+        client.load_table_from_dataframe(
+            chunk, destination_table_ref, job_config=job_config, location=location,
+        ).result()
+
+
+def load_csv_from_file(
+    client: bigquery.Client,
+    dataframe: pandas.DataFrame,
+    destination_table_ref: bigquery.TableReference,
+    location: Optional[str],
+    chunksize: Optional[int],
+    schema: Optional[Dict[str, Any]],
+):
```

Comment:
There are a lot of common parts with the `load_csv()` function, we can probably extract the common parts out. The difference seems to be just in the extra check if `schema` is `None`, and the actuall for loading a single chunk.

Reply:
I did a bit of refactoring. Indeed, they have more in common than different.